python/tank/util/shotgun/publish_resolve.py

```diff
@@ -79,41 +79,88 @@ def resolve_publish_path(tk, sg_publish_data):
     Local file storage mappings in Shotgun are global and affect
     all machines, all users and all projects.
 
-    .. note:: If no storage has been defined for the current operating system,
+    If a local storage doesn't define the operating system platform
+    you are currently using, you can augment it by adding an
+    environment variable. For example, if there
+    is a local storage named ``Renders`` that is defined on Linux and Windows,
+    you can extend to support mac by creating an environment variable named
+    ``SHOTGUN_PATH_MAC_RENDERS``. The general syntax for this is
+    ``SHOTGUN_PATH_WINDOWS|MAC|LINUX_STORAGENAME``.
+
+    .. note:: If you define a ``SHOTGUN_PATH_MAC_RENDERS`` environment variable
+              and there is a local storage ``Renders`` with a mac path set,
+              the local storage value will be used and a warning will be logged.
+
+    .. note:: If no storage can be resolved for the current operating system,
               a :class:`~sgtk.util.PublishPathNotDefinedError` is raised.
 
+
     **Resolving file urls**
 
-    The method also supports the resolution of ``file://`` urls. Such urls
-    are not multi platform and the local storages and environment variables
-    will therefore be used to try to resolve such paths in case of ambiguity.
+    The method also supports the resolution of ``file://`` urls. Contrary to
+    local file storage objects in Shotgun (see above), these paths are not
+    stored in a multi-os representation but are just defined on the operating
+    system where they were created.
+
+    If you are trying to resolve a ``file://`` url on an os platform different
+    from the one where where the url was created, the method will attempt to
+    resolve it into a valid path using several different approaches:
+
+    - First, it will look for the three environment variables ``SHOTGUN_PATH_WINDOWS``,
+      ``SHOTGUN_PATH_MAC``, and ``SHOTGUN_PATH_LINUX``. If these are defined,
+      the method will attempt to translate the path this way. For example, if
+      you are trying to resolve ``file:///prod/proj_x/assets/bush/file.txt`` on
+      windows, you could set up ``SHOTGUN_PATH_WINDOWS=P:\prod`` and
+      ``SHOTGUN_PATH_LINUX=/prod`` in order to hint the way the path should be
+      resolved.
 
-    - First, local storage settings will be downloaded from Shotgun and added to
-      a path translation table.
+    - If you wanted to use more than one environment variable, this is possible
```

Comment:
How exactly is this applied? Is "SECONDARY" a special string, or can it be anything? If anything, and if there are multiples of them, what order are they applied in?

I probably need an example to fully understand.